runtime-tests/index.ts

```diff
@@ -0,0 +1,39 @@
+const config = {
+	chain: 'polkadot',
+};
+const argv = process.argv.slice(0, 2);
+
+/**
+ * We return empty string's as an indication of a false value. JS recognizes "" as falsey.
+ * NodeJS.Process.argv.reduce expects the return value of the accumulator to
+ * be of type string, this is a way of satisfying the compiler, but also
+ * returning the correct behavior.
+ */
```

Comment:
Not sure I understand what this does. What is the expected input here? It walks all cli arguments, filters out anything that starts with "/", sets the `config[cmd]` as it goes? How is it used?

Reply:
Yea this can be a bit confusing. 

1. Parses out all inputs that start with `/`. (This is so we dont save things such as the path of the node.js binary into the config)

2. For everything else If the input doesn't start with `--`, then we return `""` (This sets the accumulator in this case `cmd` to false). This will resolve to a falsey value which will stop the reducer from saving it into the config. 

3. `argv.push(arg)` is used to save any local config options in order to pass it into the jest binary such as `--config=./runtime-tests/jest.config.js`. 

4. If the `arg` is equal to `--` and has the name of `chain` then it will set it equal to the accumulator or `cmd` and then the arg the follows will be the argument for the command itself.

Reply:
Hmm, I wonder if we're trying to be a bit too clever for our own good here; feels like this should be easier to do? ("no it's just you being thick" is a valid answer here!).

I was actually wondering something more basic: what is this args processing for? Is this right (or close to right)? 

"Process the command line arguments used to launch the test session. Merge any config options provided at the command line with the on disk config."

Reply:
Yea so you actually make a great point and I was able to incredibly simplify it. This method is way too clever for what it is suppose to be. I basically just wrote out a custom parser which doesn't need to be done either. 

I chose to do it this way because there is no straight forward way to pass custom CLI commands to jest other than setting an entry file, and calling the binary after adding a custom configuration to the disk. But that being said I also could have just used a arg-parser and then directly pass it in same as I did. Which is exactly what I changed it too. I added `argparse` now and have it just be the same as any other node argument parsing tool. No loops, or complicated logic anymore.

Good catch. Thanks